### PR TITLE
[foundation] Add support for NSValue/CGAffineTransform and StoreValueAtAddress overload

### PR DESCRIPTION
**Pull request by spouliot**
Support for `NSValue`/`CGAffineTransform` exists in iOS/tvOS/watchOS,
from UIKit, but not for macOS. However this will be required for the
new protocols inside CoreImage.

Also add an overload for `StoreValueAtAddress` since the original
one was deprecated but we did not provide the new alternative to
update the code.

### CODE REVIEW

File: src/Foundation/NSValue.cs [hunk 1]
```diff
@@ -64,6 +68,31 @@ public static NSValue FromPointF (PointF point)
...
 #endif
+
+#if MONOMAC
+		// @encode(CGAffineTransform) -> "{CGAffineTransform=dddddd}" but...
+		// using a C string crash on macOS (while it works fine on iOS)
+		[DllImport ("__Internal")]
+		extern static IntPtr xamarin_encode_CGAffineTransform ();
+
+		// The `+valueWithCGAffineTransform:` selector comes from UIKit and is not available on macOS
+		public unsafe static NSValue FromCGAffineTransform (CGAffineTransform tran)
```
Comment (rolfbjarne) [inline src/Foundation/NSValue.cs]: Style: complete word as parameter name - `transform`

Comment (spouliot) [reply src/Foundation/NSValue.cs]: yeah, I thought about it - but it's the current name for iOS/tvOS/watchOS :(
so I decided to be consistent

**Comment by spouliot** [comment]
Looks like mono's `sn` failed
```
16:32:22 ERROR: Unknown error during processing: System.NullReferenceException: Object reference not set to an instance of an object
16:32:22   at Mono.Security.StrongName.Sign (System.String fileName) [0x0001c] in <671a0913fb8b438bb20784f6952937df>:0 
16:32:22   at Mono.Tools.SN.ReSign (System.String assemblyName, System.Security.Cryptography.RSA key, System.Boolean quiet) [0x00066] in <7baa683e831147619d6d60cc01446d70>:0 
16:32:22   at Mono.Tools.SN.Process (System.String[] args) [0x0068d] in <7baa683e831147619d6d60cc01446d70>:0 
16:32:22   at Mono.Tools.SN.Main (System.String[] args) [0x00013] in <7baa683e831147619d6d60cc01446d70>:0 
16:32:22 make[10]: *** [../../../class/lib/xammac_net_4_5/nunitlite.dll] Error 1
16:32:22 make[9]: *** [do-all] Error 2
16:32:22 make[8]: *** [all-recursive] Error 1
16:32:22 make[7]: *** [../../build/deps/nunit-xammac_net_4_5.stamp] Error 2
16:32:22 make[7]: *** Waiting for unfinished jobs....
```

**Comment by spouliot** [comment]
build

**Pull request merged**